sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs

```diff
@@ -289,58 +287,66 @@ public async ValueTask<string> GetHeaderValueAsync(HttpMessage message, TokenReq
             private (TaskCompletionSource<HeaderValueInfo> InfoTcs, TaskCompletionSource<HeaderValueInfo>? BackgroundUpdateTcs, bool GetTokenFromCredential)
                 GetTaskCompletionSources(TokenRequestContext context)
             {
+                var localState = _state;
```

Comment:
Reading a three field struct isn't atomic:
```cs
internal class Program
{
    private readonly object _syncObj = new object();
    private ThreeFields _str;

    static void Main(string[] args)
    {
        new Program().Start();
    }

    private void Start()
    {
        for (var i = 0; i < Environment.ProcessorCount - 1; i++)
        {
            ThreadPool.QueueUserWorkItem(Read, null);
        }
        ThreadPool.QueueUserWorkItem(Write, null);
        Console.ReadLine();
    }

    private void Read(object state)
    {
        while (true)
        {
            var str = _str;
            Validate(str);
        }
    }

    private void Write(object state)
    {
        while (true)
        {
            lock (_syncObj)
            {
                var count = _str.FieldOne + 1;
                _str = new ThreeFields(count, count, count);
            }
        }
    }

    private static void Validate(ThreeFields str)
    {
        if (str.FieldOne != str.FieldTwo || str.FieldOne != str.FieldThree || str.FieldTwo != str.FieldThree)
        {
            Console.WriteLine($"Field one: {str.FieldOne}, Field two: {str.FieldTwo}, Field three: {str.FieldThree}");
        }
    }

    private readonly struct ThreeFields
    {
        public ThreeFields(int fieldOne, int fieldTwo, int fieldThree)
        {
            FieldOne = fieldOne;
            FieldTwo = fieldTwo;
            FieldThree = fieldThree;
        }

        public int FieldOne { get; }
        public int FieldTwo { get; }
        public int FieldThree { get; }
    }
}
```

Reply:
Thanks for the example will change to a class to get atomicity